### PR TITLE
Feature/repo implementations

### PR DESCRIPTION
**Pull request by nwokafor-choongsaeng**
Repo functions implementation

### CODE REVIEW

File: src/Morphir/IR/Repo.elm [hunk 1]
```diff
@@ -131,7 +161,42 @@ orderElmModulesByDependency parsedModules =
...
+                                    CustomTypeDeclaration typ ->
+                                        typ.name |> Node.value |> Just
+
+                                    AliasDeclaration typeAlias ->
+                                        typeAlias.name |> Node.value |> Just
+
+                                    _ -> Nothing
+                        in
+                        dec
+                            |> typeNameFromDeclaration
```
Comment (AttilaMihaly) [inline src/Morphir/IR/Repo.elm:193]: Can you please simplify this as discussed?

**Review by AttilaMihaly — CHANGES_REQUESTED** [review]

**Review by AttilaMihaly — APPROVED** [review]

File: src/Morphir/IR/Repo.elm [hunk 2]
```diff
@@ -131,14 +168,205 @@ orderElmModulesByDependency parsedModules =
...
+                                    |> Name.fromString
+                                , Type.TypeAliasDefinition typeParams tpe
+                                )
+                            )
+                        |> Just
+
+                _ ->
+                    Nothing
+    in
     Debug.todo "implement"
```
Comment (AttilaMihaly) [inline src/Morphir/IR/Repo.elm:305]: Don't you want to return `typeNameToDefinition` here?

**Review by AttilaMihaly — CHANGES_REQUESTED** [review]

File: src/Morphir/IR/Repo.elm (hunk 2 above)
Comment (nwokafor-choongsaeng) [reply src/Morphir/IR/Repo.elm]: Done. Don't know how I managed to miss that.

**Comment by nwokafor-choongsaeng** [comment]
I've added the codec in as well

**Review by AttilaMihaly — APPROVED** [review]

**Pull request merged**